src/ucs/datastruct/ptr_array.c

```diff
@@ -134,10 +143,10 @@ static void ucs_ptr_array_grow(ucs_ptr_array_t *ptr_array UCS_MEMTRACK_ARG)
     memcpy(new_array, ptr_array->start, curr_size * sizeof(ucs_ptr_array_elem_t));
 
     /* Link all new array items */
+    size_free_ahead = new_size - curr_size;
     for (i = curr_size; i < new_size; ++i) {
-        new_array[i] = UCS_PTR_ARRAY_FLAG_FREE;
-        ucs_ptr_array_placeholder_set(&new_array[i], ptr_array->init_placeholder);
-        ucs_ptr_array_freelist_set_next(&new_array[i], i + 1);
+        ucs_ptr_array_freelist_element_set(&new_array[i], size_free_ahead--,
```

Comment:
why need size_free_ahead , can we use "new_size - i" ?

Reply:
Fixed.